inc/class-command.php

```diff
@@ -259,30 +236,124 @@ protected function shell( InputInterface $input, OutputInterface $output ) {
 	}
 
 	/**
-	 * Get the name of the project host names from the local config.
+	 * Command to update the config.local.yaml file and re-provision.
+	 */
+	protected function provision( InputInterface $input, OutputInterface $output ) {
+		$success = $this->write_config_file();
+		if ( ! $success ) {
+			$output->writeln( '<error>Could not write Chassis config</error>' );
+			return 1;
+		}
+		return $this->run_command( 'vagrant provision' );
+	}
+
+	/**
+	 * Get the root directory path for the project.
+	 *
+	 * @return string
+	 */
+	protected function get_root_dir() : string {
+		return dirname( $this->getComposer()->getConfig()->getConfigSource()->getName() );
+	}
+
+	/**
+	 * Get the Local Chassis config from composer.json.
 	 *
 	 * @return array
 	 */
-	protected function get_project_hosts() : array {
+	protected function get_config() : array {
+		// @codingStandardsIgnoreLine
+		$json = file_get_contents( $this->get_root_dir() . DIRECTORY_SEPARATOR . 'composer.json' );
+		$composer_json = json_decode( $json, true );
 
-		$composer_json = json_decode( file_get_contents( getcwd() . '/composer.json' ), true );
+		return (array) $composer_json['extra']['altis']['modules']['local-chassis'] ?? [];
+	}
 
-		if ( isset( $composer_json['extra']['altis']['modules']['local-chassis']['hosts'] ) ) {
-			$hosts = (array) $composer_json['extra']['altis']['modules']['local-chassis']['hosts'];
-		} else {
-			$hosts = [ basename( getcwd() ) ];
-		}
+	/**
+	 * Writes the config.local.yaml file with Altis customisations.
+	 *
+	 * @return bool Returns false if the file write fails.
+	 */
+	protected function write_config_file() : bool {
+		// Write the default config.
+		$config = [
+			'php' => '7.2',
+			'paths' => [
+				'base' => '..',
+				'wp' => 'wordpress',
+				'content' => 'content',
+			],
+			'hosts' => [
+				basename( $this->get_root_dir() ),
+			],
+			'multisite' => true,
+			'extensions' => [
+				'humanmade/platform_chassis_extension',
+			],
+			'elasticsearch' => [
+				'plugins' => [
+					'analysis-icu',
+					'ingest-attachment',
+				],
+			],
+		];
+
+		// Merge config from composer.json.
+		$overrides = $this->get_config();
+		$config = $this->merge_config( $config, $overrides );
```

Comment:
```suggestion
		$config = $this->merge_config( $config, $this->get_config( );
```

Reply:
What do you feel is the benefit to this change? Again as above splitting this up makes debugging easier when you're given line numbers in stack traces. The less a line of code does the easier it is to read and trace.